addons/website_forum/models/forum_post.py

```diff
@@ -285,18 +292,16 @@ def _search_can_view(self, operator, value):
                    LEFT JOIN res_users u ON p.create_uid = u.id
                    LEFT JOIN forum_forum f ON p.forum_id = f.id
             WHERE
-                (p.create_uid = %s and f.karma_close_own <= %s)
-                or (p.create_uid != %s and f.karma_close_all <= %s)
-                or (
-                    u.karma > 0
-                    and (p.active or p.create_uid = %s)
-                )
+                p.create_uid = %s
```

Comment:
is there any tests for that method ? (if no we can add one)

Also do we really need that SQL query ? (I know it was there before, but maybe we can clean it)

Someting like
```py
post_domain = expression.OR([
    [('create_uid', '=', user.id)],
    [('state', 'in', ('offensive', 'pending')), ('karma_moderate', '<=', user.karma)],
    [('state', '=', 'close'), ('forum_id.karma_close_all', '<=', user.karma)],
    [('state', '=', 'active'), ('create_uid.karma', '>', 0)],
])
return post_domain if operator == "=" else ['!', *post_domain]
```